pkg/upgrade/upgrade.go

```diff
@@ -400,6 +411,22 @@ func removeCsv(c client.Client, r *rest.Config) error {
 	return nil
 }
 
+func RemoveUICR(cfg *rest.Config, reousrce schema.GroupVersionResource, instanceName string, applicationNS string) error {
+	dynamicClient, err := dynamic.NewForConfig(cfg)
+	if err != nil {
+		return err
+	}
+	if err = dynamicClient.Resource(reousrce).Namespace(applicationNS).Delete(context.TODO(), instanceName, metav1.DeleteOptions{}); err != nil {
+		if apierrs.IsNotFound(err) {
+			fmt.Printf("%s does not exist in %s, nothing to prune\n", instanceName, applicationNS)
+			return nil
+		}
+		return fmt.Errorf("error deleting %s : %v", instanceName, err)
+	}
+
+	return nil
+}
```

Comment:
```suggestion
func RemoveUICR(cfg *rest.Config, resource schema.GroupVersionResource, instanceName string, applicationNS string) error {
	dynamicClient, err := dynamic.NewForConfig(cfg)
	if err != nil {
		return err
	}
	if err = dynamicClient.Resource(resource).Namespace(applicationNS).Delete(context.TODO(), instanceName, metav1.DeleteOptions{}); err != nil {
		if apierrs.IsNotFound(err) {
			fmt.Printf("%s does not exist in %s, nothing to prune\n", instanceName, applicationNS)
			return nil
		}
		return fmt.Errorf("error deleting %s : %v", instanceName, err)
	}

	return nil
}
```

Reply:
lets decide the funciton name first then i will do a new commi with all these fixes in one go